standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -3357,6 +3355,52 @@ private void truncateTableInternal(String dbName, String tableName, List<String>
       }
     }
 
+    /**
+     * Add an empty baseDir with a truncate metadatafile
+     * @param location
+     * @param writeId
+     * @param fs
+     * @throws Exception
+     */
+    private void addTruncateBaseFile(Path location, long writeId, FileSystem fs) throws Exception {
+      Path basePath = new Path(location, AcidConstants.baseDir(writeId));
+      fs.mkdirs(basePath);
+      // We can not leave the folder empty, otherwise it will be skipped at some file listing in AcidUtils
+      // No need for a data file, a simple metadata is enough
```

Comment:
Do you think it would make sense to move metaFile creation under MetaDataFile class, not sure however it should be under AcidConstants?

Reply:
added the write to the MetaDataFile and moved it out from the AcidConstants class